oauth2.go

```diff
@@ -236,13 +236,18 @@ type Requester interface {
 
 	// Sanitize returns a sanitized clone of the request which can be used for storage.
 	Sanitize(allowedParameters []string) Requester
+
+	// GetDelegatingClient returns the client that provided the original token (subject token) in a token exchange flow.
+	GetDelegatingClient() (client Client)
 }
 
 // AccessRequester is a token endpoint's request context.
 type AccessRequester interface {
 	// GetGrantType returns the requests grant type.
 	GetGrantTypes() (grantTypes Arguments)
 
+	SetDelegatingClient(client Client)
```

Comment:
Not backwards compatible to change interfaces like this.

Reply:
Yes, see my last comment in https://github.com/ory/fosite/pull/516/files#r509957103

Reply:
Extracted interface but kept it in same file/package due to otherwise resulting cyclical dependencies, as stated in previous comment.